app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpActivity.kt

```diff
@@ -1,82 +1,44 @@
 package org.jellyfin.androidtv.ui.playback.nextup
 
-import android.graphics.Bitmap
+import android.content.Intent
 import android.os.Bundle
 import androidx.fragment.app.FragmentActivity
 import androidx.leanback.app.BackgroundManager
-import com.bumptech.glide.Glide
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
-import org.jellyfin.androidtv.util.apiclient.getItem
-import org.jellyfin.apiclient.interaction.ApiClient
-import org.jellyfin.apiclient.model.dto.ImageOptions
-import org.koin.android.ext.android.inject
-import timber.log.Timber
+import org.jellyfin.androidtv.ui.playback.PlaybackOverlayActivity
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class NextUpActivity : FragmentActivity() {
-	private lateinit var fragment: NextUpFragment
-	private val apiClient: ApiClient by inject()
+	private val viewModel: NextUpViewModel by viewModel()
 
 	override fun onCreate(savedInstanceState: Bundle?) {
 		super.onCreate(savedInstanceState)
 
-		val id = intent.getStringExtra("id")
-		if (id == null) {
-			Timber.e("No id found in bundle at onCreate().")
-			finish()
-			return
+		// Observe state
+		viewModel.state.observe(this) {state ->
+			when (state) {
+				// Open next item
+				NextUpState.PLAY_NEXT -> {
+					startActivity(Intent(this, PlaybackOverlayActivity::class.java))
+					finish()
+				}
+				// Close activity
+				NextUpState.CLOSE -> finish()
+				// Unknown state
+				else -> Unit
```

Comment:
Should we log this?

Reply:
The other possibilities are:
- NO_DATA, which is an indicator that we don't know the details of the next item, everything will work as normal but no episode title or image is shown
- INITIALIZED - Which is the default value and persists when data is correctly loaded

PLAY_NEXT and CLOSE are "actions" which this listener is meant to listen for, I therefore don't think logging is required.